tiatoolbox/dataloader/wsireader.py

```diff
@@ -0,0 +1,101 @@
+"""
+WSIReader for WSI reading or extracting metadata information from WSIs
+"""
+import pathlib
+import numpy as np
+import yaml
+from PIL import Image
+
+import openslide
+
+
+class WSIReader:
+    def __init__(
+        self,
+        input_dir=pathlib.Path.cwd(),
+        file_name=None,
+        output_dir=pathlib.Path(pathlib.Path.cwd(), "output"),
+        tile_objective_value=20,
+        tile_read_size_w=5000,
+        tile_read_size_h=5000,
+    ):
+        """
+        WSI Reader class to read WSI images
+        Args:
+            input_dir: input path to WSI directory
+            file_name: file name of the WSI
+            output_dir: output directory to save the output, default=os.getcwd()/output
+            tile_objective_value: objective value at which tile is generated, default=20
+            tile_read_size_w: tile width, default=5000
+            tile_read_size_h: tile height, default=5000
+        """
+
+        self.input_dir = pathlib.Path(input_dir)
+        self.file_name = pathlib.Path(file_name).name
+        if output_dir is not None:
+            self.output_dir = pathlib.Path(output_dir, self.file_name)
+
+        self.openslide_obj = openslide.OpenSlide(
+            filename=str(pathlib.Path(self.input_dir, self.file_name))
+        )
+        self.tile_objective_value = np.int(tile_objective_value)
+        self.tile_read_size = np.array([tile_read_size_w, tile_read_size_h])
+        self.objective_power = np.int(
+            self.openslide_obj.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER]
+        )
+        self.level_count = self.openslide_obj.level_count
+        self.level_dimensions = self.openslide_obj.level_dimensions
+        self.level_downsamples = self.openslide_obj.level_downsamples
+
+    def __exit__(self):
+        self.openslide_obj.close()
+
+    def slide_info(self, save_mode=True, output_dir=None, output_name=None):
+        """
+        WSI meta data reader
+        Args:
+            save_mode: save meta information as yaml file
+            output_dir: output directory to save the meta information
+            output_name: output file name
+
+        Returns:
+            displays or saves WSI meta information
```

Comment:
From looking at the Google doctring style guide and some sphinx output I think that these should be capitalised.